### DISCUSSION TITLE
docker extension for codeserver running in a container

### DISCUSSION THREAD

**Discussion by mezoology** [coder/code-server#5454] [General] [answered]
hello, I'm struggling to successfully configure the docker extension on code-server container
I use this image:  lscr.io/linuxserver/code-server:latest

what I want to do is connect the extension to the hosts (docker.sock) and test/run containers on the host.

by host, i mean the host that code-server container is running on.

is this achievable ?

**Comment by mezoology** [comment, marked as answer]
enabled the docker mod, and it worked.

**Reply by EpickyHrac** [reply]
How did you solve the problem? Did you add an ENV variable or something else? I don't understand in "Docker Mods" I would appreciate your help

**Reply by lg-nineoneninethreenine** [reply]
Maybe they installed Docker on the Docker environment(how weird is that?)

**Reply by lg-nineoneninethreenine** [reply]
I think it works like that.

**Reply by EpickyHrac** [reply]
I have too installed in code-server container docker.io via apt, but says `Error: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/json?filters=%7B%22dangling%22%3A%7B%22false%22%3Atrue%7D%7D": dial unix /var/run/docker.sock: connect: permission denied` ![image](https://user-images.githubusercontent.com/66244226/210392088-cd968c52-0cb6-4ee9-ae5c-921638ad2923.png)

**Reply by lg-nineoneninethreenine** [reply]
Is code-server running inside Docker? If yes, then I daresay you could rather use your base system with Docker.

**Reply by EpickyHrac** [reply]
Yes code-server is running in inside Docker.

**Reply by lg-nineoneninethreenine** [reply]
Then I doubt you can use Docker inside a Docker container. @jsjoeio I haven't heard of it, but is it possible?

**Reply by jsjoeio** [reply]
Yes, it's possible! We have a concept in the product (Coder) called "container-based VMs" that is essentially this. I don't know the intricacies of it.

**Reply by lg-nineoneninethreenine** [reply]
Hmm... It's like running Python inside Python, it's just looking haywired.
